tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -0,0 +1,185 @@
+#include "tenseal/cpp/tensors/ckkstensor.h"
+
+namespace tenseal {
+
+using namespace seal;
+using namespace std;
+
+CKKSTensor::CKKSTensor(const shared_ptr<TenSEALContext>& ctx,
+                       const PlainTensor<double>& tensor,
+                       std::optional<double> scale)
+    : _shape(tensor.shape()), _strides(tensor.strides()) {
+    this->link_tenseal_context(ctx);
+    if (scale.has_value()) {
+        this->_init_scale = scale.value();
+    } else {
+        this->_init_scale = ctx->global_scale();
+    }
+
+    vector<double> plain_data = tensor.data();
+    for (int i = 0; i < plain_data.size(); i++)
+        _data.push_back(
+            CKKSTensor::encrypt(ctx, this->_init_scale, plain_data[i]));
+}
+
+Ciphertext CKKSTensor::encrypt(const shared_ptr<TenSEALContext>& ctx,
+                               const double scale, const double data) {
+    return encrypt(ctx, scale, vector<double>({data}));
+}
+
+Ciphertext CKKSTensor::encrypt(const shared_ptr<TenSEALContext>& ctx,
+                               const double scale, const vector<double>& data) {
```

Comment:
data can be a PlainTensor as well.

Reply:
The rest of the API uses PlainTensor for encrypt. Maybe we should do the same here.

Reply:
I see this encrypt more as an internal private function, that takes a vector of double values to batch into a single ciphertext. The constructor is responsible for manipulating the PlainTensor and constructing these vectors for encryption. What do you think?

Reply:
I think it would be better for consistency to work with PlainTensors, because the rest of the tensors work with it already.

Or we should update them as well, to have a similar encrypt declaration.